dxr/static_unhashed/js/dxr.js

```diff
@@ -245,6 +245,130 @@ $(function() {
     }
 
     /**
+     * Visit the result lines in result and remove duplicated lines and redundant context buttons.
+     */
+    function cleanupResults(result) {
+        // Construct {line: {hasContext, hasResult}}
+        var lineMap = {};
+        // Visit the lines and remove duplicates.
+        result.children('.result_line').each(function() {
+            var line = parseInt($(this).data("line"));
+            lineMap[line] = lineMap[line] || {};
+            if (this.classList.contains("ctx_row")) {
+                if (lineMap[line].hasContext) {
+                    // Then this is duplicate, remove it.
+                    $(this).remove();
+                } else {
+                    lineMap[line].hasContext = this;
```

Comment:
`hasContext` sounds like a boolean, not an element. Maybe this should be named `contextEl`, or something else?